src/test/java/com/cloudbees/jenkins/support/impl/GCLogsTest.java

```diff
@@ -52,6 +54,31 @@ public void rotatedFiles() throws Exception {
         assertEquals(15, container.getContents().size());
     }
 
+    @Test
+    @Issue("JENKINS-58980")
+    public void latestFiles() throws Exception {
+        File tempDir = Files.createTempDir();
+        long currentTime = System.currentTimeMillis();
+        for (int count = 0; count < 10; count++) {
+            File gcLogFile = new File(tempDir, "gc5625.log" + count);
+            Files.touch(gcLogFile);
+            gcLogFile.setLastModified(currentTime - TimeUnit.DAYS.toMillis(5));
+        }
+        for (int count = 0; count < 5; count++) {
+            Files.touch(new File(tempDir, "gc3421.log" + count));
+        }
+
+        GCLogs.VmArgumentFinder finder = mock(GCLogs.VmArgumentFinder.class);
+        when(finder.findVmArgument(GCLogs.GCLOGS_JRE_SWITCH)).thenReturn(GCLogs.GCLOGS_JRE_SWITCH + new File(tempDir, "gc%p.log").getAbsolutePath());
+        when(finder.findVmArgument(GCLogs.GCLOGS_ROTATION_SWITCH)).thenReturn(GCLogs.GCLOGS_ROTATION_SWITCH);
+
+        TestContainer container = new TestContainer();
+
+        new GCLogs(finder).addContents(container);
+
+        assertEquals(5, container.getContents().size());
```

Comment:
Maybe checking the name to be sure these are the recently modified ones?

Reply:
Thanks. Done.